Bottom: ebcff211fbf53f81588a32a8f6bef3bd23feaf41
Top:    54fcc0339be4f81b7d9c0404ee162b5b332c7f8b
Author: melsdoerfer <michael@elsdoerfer.com>
Date:   2009-11-04 08:46:42 +0100

Expose the raw feed content as an item ('data') in the result dict. This is for example useful for implementing features where you need access to data that is not parsed (e.g. comments), say for example claimcode-like functionality.

Not necessary to run FeedPlatform itself, but custom addins may find this feature useful.

Reported as issue #164.


---

diff --git a/feedparser/feedparser.py b/feedparser/feedparser.py
index 9a341f4..7ecf138 100644
--- a/feedparser/feedparser.py
+++ b/feedparser/feedparser.py
@@ -3412,7 +3412,7 @@ def parse(url_file_stream_or_string, etag=None, modified=None, agent=None, refer
         handlers = [handlers]
     try:
         f = _open_resource(url_file_stream_or_string, etag, modified, agent, referrer, handlers)
-        data = f.read()
+        data = result['data'] = f.read()
     except Exception, e:
         result['bozo'] = 1
         result['bozo_exception'] = e
diff --git a/feedparser/feedparsertest.py b/feedparser/feedparsertest.py
index d3606bb..1a93ecf 100644
--- a/feedparser/feedparsertest.py
+++ b/feedparser/feedparsertest.py
@@ -98,11 +98,7 @@ class FeedParserTestServer(Thread):
 class TestCase(unittest.TestCase):
   def failUnlessEval(self, evalString, env, msg=None):
     """Fail unless eval(evalString, env)"""
-    failure=(msg or 'not eval(%s)' % evalString)
-    try:
-      env = env.data
-    except:
-      pass
+    failure=(msg or 'not eval(%s)' % evalString)    
     if not eval(evalString, env):
       raise self.failureException, failure
